### DISCUSSION TITLE
Questions on the implementation

### DISCUSSION THREAD

**Discussion by daveajones** [skymethod/op3#2] [General]
Is this using CF workers and storing the data in CF's distributed key/value db?

**Comment by johnspurlock-skymethod** [comment]
Yes, each [deployment](https://github.com/skymethod/op3/blob/master/deploy.sh) is a Cloudflare Worker script, the logs are stored in [Durable Objects](https://developers.cloudflare.com/workers/learning/using-durable-objects/), Cloudflare's lowest-level transactional storage primitive.

I could see using [KV](https://developers.cloudflare.com/workers/learning/how-kv-works/) in the future to help out on serving API requests - it's great for things that are mostly reads

Started an [architecture doc](https://github.com/skymethod/op3/blob/master/docs/architecture.md)

**Comment by johnspurlock-skymethod** [comment]
A side benefit of using Durable Objects (DOs) for data storage in this project, is that there is no _admin backend_ api to this storage.

All reads/writes have to go through workers, and hence folks can see every access point in the public codebase here

**Comment by daveajones** [comment]
That's a really solid design John.
